buckets/redis/bucket.go

```diff
@@ -113,19 +128,39 @@ func (bf *bucketFactory) NewBucket(namespace, bucketName string, cfg *pbconfig.B
 		idle = strconv.FormatInt(int64(cfg.MaxIdleMillis), 10)
 	}
 
-	rb := &redisBucket{
-		dyn,
+	var rbc *redisBucketCommon
+
+	if dyn {
+		var exists bool
+		bf.Lock()
+		defer bf.Unlock()
```

Comment:
we need to surround this in a function or just call `bf.Unlock` at the end of the`if` block..

https://play.golang.org/p/-MBcYET03D